src/mixin.js

```diff
@@ -9,6 +9,9 @@ export default {
     options.i18n = options.i18n || (options.__i18n ? {} : null)
 
     if (options.i18n) {
+      if (typeof options.i18n === 'function') {
```

Comment:
I guess, iInstead of allowing `i18n` as factory, using same design (hooks) in Vuex as you mentioned(https://github.com/vuejs/vuex/blob/caa663d69608fc36d1a88746df28961437d38786/src/mixin.js#L5) might help users for understanding.
What do you think about design choices for this feature ?

Reply:
I don't get what you mean exactly @exoego , if you look https://github.com/vuejs/vuex/blob/caa663d69608fc36d1a88746df28961437d38786/src/mixin.js#L26 it's the same approach I used in this pull request

Reply:
Ah, sorry.
You are right that `options.i18n` in vue-i18n is analogus to `options.store` in vuex.
I thought vuex has some special handling in hook.